presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGeneratorContext.java

```diff
@@ -360,7 +377,77 @@ else if (!groupByColumns.isEmpty()) {
 
         LinkedHashMap<VariableReferenceExpression, PinotColumnHandle> assignments = getAssignments();
         List<Integer> indices = getIndicesMappingFromPinotSchemaToPrestoSchema(query, assignments);
-        return new PinotQueryGenerator.GeneratedPql(tableName, query, indices, groupByColumns.size(), filter.isPresent(), isQueryShort);
+        return new PinotQueryGenerator.GeneratedPinotQuery(tableName, null, query, indices, groupByColumns.size(), filter.isPresent(), isQueryShort);
+    }
+
+    /**
+     * Convert the current context to a Pinot SQL
+     */
+    public PinotQueryGenerator.GeneratedPinotQuery toSqlQuery(PinotConfig pinotConfig, ConnectorSession session)
```

Comment:
Didn't read this too carefully ... this looks like a giant copy paste :-). What's the main change from PQL generation ?

Reply:
Current major diff is to change `TOP` to `LIMIT`.
Here I expect the future change will happen more on SQL side than PQL side, so intentionally to separate both.
I will abstract common parts(selection/filtering) to until functions for now.

Reply:
extract the common logic to a method.
Current difference is SELECT/LIMIT clauses generated are different.